Source/Events.Store.MongoDB/CommittedEventsFetcher.cs

```diff
@@ -83,24 +86,78 @@ public async Task<CommittedEvents> FetchCommittedEvents(ScopeId scope, EventLogS
             throw new EventStoreUnavailable("Mongo wait queue is full", ex);
         }
     }
-    
+
     /// <inheritdoc/>
-    public Task<CommittedAggregateEvents> FetchForAggregate(EventSourceId eventSource, ArtifactId aggregateRoot, CancellationToken cancellationToken)
-        => DoFetchForAggregate(
+    public async Task<CommittedAggregateEvents> FetchForAggregate(EventSourceId eventSource, ArtifactId aggregateRoot, CancellationToken cancellationToken)
+    {
+        var (aggregateRootVersion, eventStream) = await DoFetchForAggregate(
             eventSource,
             aggregateRoot,
-            defaultFilter => defaultFilter,
-            cancellationToken);
+            filter => filter,
+            cancellationToken).ConfigureAwait(false);
+
+        var events = await eventStream.ToListAsync(cancellationToken);
+        if ((ulong) events.Count != aggregateRootVersion)
+        {
+            throw new InconsistentNumberOfCommittedAggregateEvents(eventSource, aggregateRoot, aggregateRootVersion, events.Count);
+        }
+        return new CommittedAggregateEvents(eventSource, aggregateRoot, aggregateRootVersion, events);
+    }
+
+    /// <inheritdoc/>
+    public async Task<Try<(AggregateRootVersion AggregateRootVersion, IAsyncEnumerable<CommittedAggregateEvent> EventStream)>> FetchStreamForAggregate(
+        EventSourceId eventSource,
+        ArtifactId aggregateRoot,
+        IEnumerable<Artifact> eventTypes,
+        CancellationToken cancellationToken)
+    {
+        try
+        {
+            return await DoFetchForAggregate(
+                eventSource,
+                aggregateRoot,
+                filter => filter & _eventFilter.In(_ => _.Metadata.TypeId, eventTypes.Select(_ => _.Id.Value)),
```

Comment:
Oh - so the filter already only does look at the `TypeId` - then it makes even more sense to only expose the `ArtifactId` on the interface :)